### PR TITLE
faster applySourceBlocks

### PR DESCRIPTION
**Pull request by hjanetzek**

### CODE REVIEW

File: core/src/gl/shaderProgram.cpp [hunk 1]
```diff
@@ -281,34 +278,39 @@ std::string ShaderProgram::applySourceBlocks(const std::string& source, bool fra
...
-        it += sm.position() + sm.length();
+    while (std::getline(sourceIn, line)) {
+        if (line.empty()) {
+            continue;
+        }
 
-        std::string pragmaName = sm[1];
-        bool unique;
-        std::tie(std::ignore, unique) = pragmas.emplace(std::move(pragmaName));
+        sourceOut << line << '\n';
```
Comment (karimnaaji) [inline core/src/gl/shaderProgram.cpp]: Just tested, and this is definitely an improvement!

Do you think we can apply the "Replace blackslash+newline with spaces" part at this point on `line` variable.
If we find `\\` on `line` variable directly and not inject `\n` on the stream at this point?

File: core/src/gl/shaderProgram.cpp [hunk 2]
```diff
@@ -286,7 +294,15 @@ std::string ShaderProgram::applySourceBlocks(const std::string& source, bool fra
             continue;
         }
 
-        sourceOut << line << '\n';
+        size_t start = 0;
+
+        // Replace blackslash+newline with spaces
+        if ((start = line.find("\\", start)) != std::string::npos) {
+            line.replace(start, 1, " ");
```
Comment (hjanetzek) [inline core/src/gl/shaderProgram.cpp]: Better check the last character explicitly

**Comment by karimnaaji** [comment]
We could also strip line comments from our script that builds the shaders to header files.

**Comment by blair1618** [comment]
@karimnaaji @hjanetzek Are you still working on this? Seems pretty good to me.

**Comment by karimnaaji** [comment]
@blair1618 , we spoke about adding one more commit to remove all our `\` from `core/shaders` and making sure we don't use them anywhere else than `rasters.glsl`. 
Coming shortly!

**Pull request merged**